File: data/gpt-4o/readable_format_HPEC/cpp/file238.cpp
void init_array(int maxgrid, float sumTang[100][100], float mean[100][100], float path[100][100])
{
 for(int i=0; i<maxgrid; i++)
 for(int j=0; j<maxgrid; j++)
 {
  sumTang[j][i] = i*j;
  mean[j][i] = (float)(i-j)/maxgrid;
  path[j][i] = (float)((i-1)*(j-2))/maxgrid;
 }
}

File: data/llama3_8B_ins_results/readable_format_HPEC/fortran/file238.f90
subroutine init_array(maxgrid, sumTang, mean, path)
implicit none

integer :: maxgrid
DATA_TYPE, dimension (maxgrid, maxgrid) :: sumTang, mean, path
integer :: i, j
do i = 1, maxgrid
do j = 1, maxgrid
sumTang(j, i) = i * j
mean(j, i) = ( i - j ) / (maxgrid)
path(j, i) = (( i - 1 ) * ( j - 2 )) / (maxgrid)
end do
end do
end subroutine
